cpp/ruapu.hpp

```diff
@@ -0,0 +1,39 @@
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#define RUAPU_IMPLEMENTATION
+#include "../ruapu.h"
+
+class ruapu
+{
+public:
+    ruapu() { 
+      static int initialised = 0;
+      if (!initialised) {
+        ruapu_init();
+        initialised = 1;
+      }
+    }
+
+    ~ruapu() = default;
+
+    bool support(std::string isa)
+    {
+        return ruapu_supports(isa.c_str()) == 1;
+    }
+
+    std::vector<std::string> rua()
+    {
+        std::vector<std::string> supported_isa_vec;
```

Comment:
Initialise ruapu before using any of its APIs.

```suggestion
        ruapu::init();
        std::vector<std::string> supported_isa_vec;
```